src/python/sdk/econia_sdk/utils/decimals.py

```diff
@@ -0,0 +1,67 @@
+import math
+
+def get_lot_size(smallest_decimal_unit: float, coin_decimals: int) -> int:
+    """
+    Given a decimal representation of the smallest possible unit and the
+    whole unit's decimals (e.g. ETH has 18 decimals), return the lot size.
+
+    Parameters:
+    * `smallest_decimal_unit`: Decimal (e.g. 0.001 for one thousandth) of
```

Comment:
Python docs themselves list some issues with using floats for base-10 financial applications, so I'd advise using a Python decimal under the hood

Here, have the user pass arguments as a string, e.g. `'0.0001'`

Then inside the function first assert that the input is a string, and convert the string to a decimal.Decimal

This should do away with potential errors

Reply:
Also then do this for all other inputs that are not strictly integers